03-task-manager/starter/controllers/tasks.js

```diff
@@ -0,0 +1,58 @@
+const Task = require("../models/Task");
+const asyncWrapper = require("../middleware/async");
+const { createCustomError } = require("../errors/custom-error");
+
+const getAllTasks = asyncWrapper(async (req, res) => {
+  const tasks = await Task.find({});
+  res.status(200).json({ tasks });
+});
+
+const createTask = asyncWrapper(async (req, res) => {
+  const task = await Task.create(req.body);
+  res.status(201).json({ task });
+});
+
+const getTask = asyncWrapper(async (req, res, next) => {
+  const { id: taskID } = req.params;
+  const task = await Task.findOne({ _id: taskID });
+
+  if (!task) {
+    return next(createCustomError(`No Task With ID: ${taskID}`, 404));
+  }
+
+  res.status(200).json({ task });
+});
+
+const updateTask = asyncWrapper(async (req, res) => {
```

Comment:
Don't forget to pass `next` whenever you actually use it, like below in the if clause checking for the existence of the task. (This feedback is also relevant in the `deleteTask` handler.)